code/model/UserDefinedForm.php

```diff
@@ -537,46 +537,43 @@ public function Form() {
 	 */
 	public function getFormFields() {
 		$fields = new FieldList();
-				
-		if($this->Fields()) {
-			foreach($this->Fields() as $editableField) {
-				// get the raw form field from the editable version
-				$field = $editableField->getFormField();
-				if(!$field) break;
-				
-				// set the error / formatting messages
-				$field->setCustomValidationMessage($editableField->getErrorMessage());
 
-				// set the right title on this field
-				if($right = $editableField->getSetting('RightTitle')) {
-					$field->setRightTitle($right);
-				}
-				
-				// if this field is required add some
-				if($editableField->Required) {
-					$field->addExtraClass('requiredField');
-					
-					if($identifier = UserDefinedForm::config()->required_identifier) {
-						
-						$title = $field->Title() ." <span class='required-identifier'>". $identifier . "</span>";
-						$field->setTitle($title);
-					}
-				}
-				// if this field has an extra class
-				if($editableField->getSetting('ExtraClass')) {
-					$field->addExtraClass(Convert::raw2att(
-						$editableField->getSetting('ExtraClass')
-					));
-				}
-				
-				// set the values passed by the url to the field
-				$request = $this->getRequest();
-				if($var = $request->getVar($field->name)) {
-					$field->value = Convert::raw2att($var);
+		$editableFields = $this->Fields();
+		if($editableFields) foreach($editableFields as $editableField) {
+			// get the raw form field from the editable version
+			$field = $editableField->getFormField();
+			if(!$field) break;
+
+			// set the error / formatting messages
+			$field->setCustomValidationMessage($editableField->getErrorMessage());
+
+			// set the right title on this field
+			if($right = $editableField->getSetting('RightTitle')) {
+				$field->setRightTitle(Convert::raw2xml($right));
```

Comment:
Can you add a comment here why we escape? Something along the lines of "The FormField implementation allows HTML, but since this is user data we filter it"